### DISCUSSION TITLE
How to disable `<space>` command when in VSCode?

### DISCUSSION THREAD

**Discussion by macintacos** [LazyVim/LazyVim#3558] [Q&A]
I'm trying to use the VSCode "extras" configuration, and in VSCode I want to use [VSpaceCode](https://vspacecode.github.io/). I want to invoke VSpaceCode with a single invocation of the `<space>` key, as one would normally expect. So to do that, I have the following code:

```lua
if vim.g.vscode then
  -- In VSCode, don't delete mappings that don't matter.
  local vscode = require("vscode")

  vim.keymap.set({ "n", "x", "v" }, "<space>", function()
    vscode.call("vspacecode.space")
  end)
end
```

This works, but there is a noticeable delay (have to press space, then wait a second, and then VSpaceCode's menu appears).

https://github.com/LazyVim/LazyVim/assets/8128941/de6b8b9a-824d-41ae-9eb1-b60555258414

Funnily enough, it seems like if I hit space twice quickly, it displays the list of files to search for in VSCode itself.

https://github.com/LazyVim/LazyVim/assets/8128941/7ff740c0-71e8-4d12-8ba4-b1a2a261a727

My current leading theory is that the first time I hit `<space>` in VSCode, it actually waits for a timeout to make sure I'm not trying to hit `<space>` again, which in the default LazyVim configuration is the command to search all files.

How do I get the `<space><space>` keymap disabled when running VSCode with `vscode-neovim`? Right now when I run `vim.keymap.del("n", "<leader><leader>")` or `vim.keymap.del("n", "<space><space>")`, I get an error from `vscode-neovim` saying that no such mapping exists. I assuming this is probably because the keymaps are lazy loaded or something like that, but I am not quite sure how to go about overriding that behavior.

**Comment by macintacos** [comment]
Setting `timeoutlen` to `0` seems to have done the trick. But won't that mess other things up? Or is that fine as long as I gate it as only being enabled when I'm in VSCode?

**Comment by folke** [comment]
```lua
vim.keymap.set({ "n", "x", "v" }, "<space>", function()
    vscode.call("vspacecode.space")
  end, {nowait = true})
```

**Reply by macintacos** [reply]
Unfortunately that doesn't seem to fix the problem. When I disable the `timeoutlen` change that I made add `{ nowait = true }` to the keymap, there is still a noticeable delay, about 1 second by my count.

I just pushed my configuration here, if you want to point out anything I'm doing wrong: https://github.com/macintacos/nvim
